File: spring-auto-restdocs-example/src/test/java/capital/scalable/restdocs/example/DocumentationTest.java
package capital.scalable.restdocs.example;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.forwardedUrl;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

import capital.scalable.restdocs.example.testsupport.MockMvcBase;
import org.junit.Test;

public class DocumentationTest extends MockMvcBase {

    @Test
    public void docsForwarding() throws Exception {
        mockMvc.perform(get("/docs"))
                .andExpect(status().isOk())
                .andExpect(forwardedUrl("/docs/index.html"));
    }
}

File: samples/kotlin-webmvc/src/test/kotlin/capital/scalable/restdocs/example/DocumentationTest.kt
/*-
 * #%L
 * Spring Auto REST Docs Kotlin Web MVC Example Project
 * %%
 * Copyright (C) 2015 - 2021 Scalable Capital GmbH
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
package capital.scalable.restdocs.example

import capital.scalable.restdocs.example.testsupport.MockMvcBase
import org.junit.jupiter.api.Test
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.forwardedUrl
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status

class DocumentationTest : MockMvcBase() {

    @Test
    fun docsForwarding() {
        mockMvc.perform(get("/docs"))
            .andExpect(status().isOk)
            .andExpect(forwardedUrl("/docs/index.html"))
    }
}
